hack/ensure_job_cluster.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+"""This script ensures the cluster of prowjobs is defined as expected.
+e.g., python ./hack/ensure_job_cluster.py
+"""
+
+import argparse
+import logging
+import os
+import sys
+import yaml
+
+
+def get_desired_cluster(file_path, job):
+    """get the desired cluster for a given job defined in file_path."""
+    if job.get("agent", "") != "kubernetes":
+        return ""
+    if job["name"] in JOB_MAP:
+        return JOB_MAP[job["name"]]
+    if job["name"].endswith("-build01") or migrated(file_path):
+        return BUILD01_CLUSTER
+    return ""
+
+
+def handle(file_path, jobs):
+    """handle jobs."""
+    new_jobs = []
+    names = []
+    for job in jobs:
+        cluster = get_desired_cluster(file_path, job)
+        if cluster != job.get("cluster", ""):
+            logging.error("expected the cluster of job %s to be %s, but found %s",
+                          job["name"], cluster, job.get("cluster", ""))
+            names.append(job["name"])
+            job["cluster"] = cluster
+        new_jobs.append(job)
+    return new_jobs, names
+
+
+def migrated(file_path):
+    """check if the jobs defined in file_path are migrated."""
+    # we do not migrate the periodis in release repo
+    # due to https://github.com/openshift/release/pull/7178
+    if file_path.endswith('periodics.yaml') and 'openshift/release/' in file_path:
+        return False
+    return False
+
+def ensure(job_dir, overwrite):
+    """ensure prow jobs' cluster."""
+    i = 0
+    names = []
+    for dirpath, _, filenames in os.walk(job_dir):
+        for filename in filenames:
+            if filename.endswith('.yaml'):
+                file_path = os.path.join(dirpath, filename)
+                i += 1
+                if i % 100 == 1:
+                    logging.info("handling file %d: %s", i, file_path)
+                with open(file_path) as file:
+                    data = yaml.safe_load(file)
+                    for job_type in ["presubmits", "postsubmits"]:
+                        if job_type not in file_path:
+                            continue
+                        for repo in data[job_type]:
+                            jobs, tmp_names = handle(file_path, data[job_type][repo])
+                            names.extend(tmp_names)
+                            data[job_type][repo] = jobs
+                    if "periodics" in file_path:
```

Comment:
Why assume that a file will have this in the path? What if someone makes a repo called `periodics`? You can just operate on `data.get("periodics", [])`

Reply:
Done.